joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala

```diff
@@ -1,130 +1,755 @@
+        |import java.util.ArrayList;
+        |import java.util.List;
+        |import java.util.stream.Collectors;
+        |
+        |public class TestClass {
+        |  public Integer method(Integer aaa) {
+        |    List<Integer> list = new ArrayList<>();
+        |    list.add(1);
+        |
+        |    List<Integer> mappedList = list.stream().map(integer -> {
+        |      List<Integer> nestedList = new ArrayList<>();
+        |      nestedList.add(1);
+        |
+        |      List<Integer> nestedMappedList =
+        |          nestedList.stream().map(nestedInteger -> nestedInteger + aaa).collect(Collectors.toList());
+        |      return nestedMappedList.get(0);
+        |    }).collect(Collectors.toList());
+        |    Integer ret = mappedList.get(0);
+        |    return ret;
+        |  }
+        |}
+        |""".stripMargin)
+
+    "create 2 method nodes for the respective lambdas" in {
+      cpg.method.name(".*lambda.*").l match {
+        case List(lambda0, lambda1) =>
+          lambda0.name shouldBe "lambda$0"
+
+          lambda1.name shouldBe "lambda$1"
+
+        case result => fail(s"Expected 2 lambda methods but got $result")
+      }
     }
   }
 
-  "CPG for code with a simple lambda which captures a local" - {
-    lazy val cpg = JavaSrc2CpgTestContext.buildCpg("""
-        |import java.util.Comparator;
+  "lambdas used as a function argument" should {
+    val cpg = code("""
+        |import java.util.function.Function;
         |
         |public class Foo {
+        |  public static String getFromSupplier(String input, Function<String, String> mapper) {
+        |    return mapper.apply(input);
+        |  }
         |
-        |    int value;
+        |  public void test1(String input, String fallback) {
+        |    getFromSupplier(
+        |      input,
+        |      lambdaInput -> lambdaInput.length() > 5 ? "Long" : fallback
+        |    );
+        |  }
+        |}
+        |""".stripMargin)
+
+    "create a method node for the lambda" in {
+      cpg.typeDecl.name("Foo").method.name(".*lambda.*").l match {
+        case List(lambdaMethod) =>
+          // Lambda body creation tested separately
+          lambdaMethod.name shouldBe "lambda$0"
+          lambdaMethod.fullName shouldBe "Foo.lambda$0:java.lang.String(java.lang.String)"
+          lambdaMethod.parameter.l match {
+            case List(lambdaInput) =>
+              lambdaInput.name shouldBe "lambdaInput"
+              lambdaInput.typeFullName shouldBe "java.lang.String"
+
+            case result => fail(s"Expected single lambda parameter but got $result")
+          }
+          lambdaMethod.methodReturn.typeFullName shouldBe "java.lang.String"
+
+        case result => fail(s"Expected single lambda method but got $result")
+      }
+    }
+
+    "create the correct binding for the lambda method" in {
+      cpg.all.collectAll[Binding].filter(_.name == "lambda$0").l match {
+        case List(lambdaBinding) =>
```

Comment:
There should not be a binding with the name of the implementing method. Just the bindings of the implemented methods name is enough. It does not really hurt but it is noise so please remove those bindings.